### PR TITLE
Log input devices

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Drops some unused code from the "Input Report", and replaces the optional logging with direct logging.

This now includes logging the actual configuration of the device

### CODE REVIEW

**Review by Saviq — APPROVED** [review]
Couple of nits, but feel free to merge if you don't think worth it.

---

While here, should we unify the logging categories?

https://github.com/canonical/mir/blob/31ef0787e794f23e1eca415f575cd5859ec096b4/src/server/input/default_input_device_hub.cpp#L30

Could be `input-hub` instead?

```
 <information> evdev-input: Opened device: /dev/input/event9 [0000:0006] Video Bus
 <information> Input: Device configuration: Lid Switch, capabilities={}
 <information> evdev-input: Opened device: /dev/input/event0 [0000:0005] Lid Switch
 <information> Input: Device configuration: Power Button, capabilities={keyboard}
 <information> evdev-input: Opened device: /dev/input/event1 [0000:0001] Power Button
#  …
```

---

Looks like an opportunity to introduce `std::formatter<>`s for the different types?

**Comment by AlanGriffiths** [comment]
> Looks like an opportunity to introduce `std::formatter<>`s for the different types?

Oh! Yeah, I still have to get used to having this stuff available.

File: src/server/input/default_input_device_hub.cpp [hunk 1]
```diff
@@ -195,6 +196,264 @@ mi::DefaultInputDeviceHub::DefaultInputDeviceHub(
...
+        case mir_pointer_acceleration_adaptive:
+            out = std::format_to(out, "acceleration=adaptive,");
+            break;
+
+        case mir_pointer_acceleration_none:
+            out = std::format_to(out, "acceleration=none,");
+            break;
+        }
+
+        out = std::format_to(out, "acceleration={:.2f},", c.cursor_acceleration_bias());
```
Comment (Saviq) [inline src/server/input/default_input_device_hub.cpp:311]:
```suggestion
        out = std::format_to(out, "acceleration-bias={:.2f},", c.cursor_acceleration_bias());
```

**Review by Saviq — COMMENTED** [review]
Should we align the command line options with what's logged?

`--mouse-acceleration` vs. `capabilities={pointer}…, pointer_config={acceleration=…}`

Underscores vs. hyphens, too?

**Review by Saviq — CHANGES_REQUESTED** [review]
Still no…

```
$ ubuntu-frame --mouse-acceleration=none --mouse-cursor-acceleration-bias=1.00 2>&1 | grep -e input-hub -e evdev-input
[2024-06-19 08:09:21.795207] <information> mirserver: Selected input driver: mir:evdev-input (version: 2.18.0)
[2024-06-19 08:09:25.329572] <information> evdev-input: Opened device: /dev/input/event3 [0000:0001] Power Button
[2024-06-19 08:09:25.351274] <information> evdev-input: Opened device: /dev/input/event7 [0000:0006] Video Bus
[2024-06-19 08:09:25.373068] <information> evdev-input: Opened device: /dev/input/event1 [0000:0001] Power Button
[2024-06-19 08:09:25.373404] <information> evdev-input: Opened device: /dev/input/event0 [0000:0005] Lid Switch
[2024-06-19 08:09:25.395651] <information> evdev-input: Opened device: /dev/input/event2 [0000:0003] Sleep Button
[2024-06-19 08:09:25.417728] <information> evdev-input: Opened device: /dev/input/event5 [05ac:024a] Apple Inc. Apple Internal Keyboard / Trackpad
[2024-06-19 08:09:25.419174] <information> evdev-input: Opened device: /dev/input/event8 [05ac:024a] bcm5974
[2024-06-19 08:09:25.419356] <information> input-hub: Device configuration: Power Button, capabilities={keyboard}
[2024-06-19 08:09:25.419481] <information> input-hub: Device configuration: Video Bus, capabilities={keyboard}
[2024-06-19 08:09:25.419586] <information> input-hub: Device configuration: Power Button, capabilities={keyboard}
[2024-06-19 08:09:25.419687] <information> input-hub: Device configuration: Lid Switch, capabilities={}
[2024-06-19 08:09:25.419787] <information> input-hub: Device configuration: Sleep Button, capabilities={keyboard}
[2024-06-19 08:09:25.432277] <information> input-hub: Device configuration: Apple Inc. Apple Internal Keyboard / Trackpad, capabilities={keyboard|alpha_numeric}
[2024-06-19 08:09:25.432513] <information> input-hub: Device configuration: bcm5974, capabilities={pointer|touchpad}, pointer_config={handedness=right,acceleration=adaptive,acceleration_bias=0.00,hscale=1.00,vscale=1.00}, touchpad_config={click_mode=finger_count|scroll_mode=two_finger|disable_while_typing}
```

**Comment by Saviq** [comment]
(but you will need to fix the tests…)

**Review by Saviq — APPROVED** [review]
Oh well, `sid`'s got a short run…

**Pull request merged**